packages/editor/src/editor.ts

```diff
@@ -0,0 +1,207 @@
+    this._input = input;
+    this._vscodeDir = vscodeDir;
+    this._tmpDir = tmpDir;
+    this._instanceState = instanceState;
+    this._makeMultilineJSIntoSingleLine = makeMultilineJSIntoSingleLine;
+    this._loadExternalCode = loadExternalCode;
+    this._content = '';
+    this.print = instanceState.context.print;
+
+    // Add edit command support to shell api.
+    const wrapperFn = (...args: string[]) => {
+      return Object.assign(this.runEditCommand(args), {
+        [Symbol.for('@@mongosh.syntheticPromise')]: true
+      });
+    };
+
+    wrapperFn.isDirectShellCommand = true;
+    wrapperFn.returnsPromise = true;
+    (instanceState.shellApi as any).edit = instanceState.context.edit = wrapperFn;
+    (signatures.ShellApi.attributes as any).edit = {
+      type: 'function',
+      returnsPromise: true,
+      isDirectShellCommand: true
+    } as TypeSignature;
+  }
+
+  static create(options: EditorOptions): Editor {
+    return new Editor(options);
+  }
+
+  // In the case of using VSCode as an external editor,
+  // detect whether the MongoDB extension is installed and open a .mongodb file.
+  // In all other cases open a .js file.
+  async _getExtension(cmd: string): Promise<string> {
+    if (!this._isVscodeApp(cmd)) {
+      return 'js';
+    }
+
+    try {
+      const extensions = await fs.readdir(path.join(this._vscodeDir, 'extensions'));
+      const hasMongodbExtension = !!extensions
+        .find((name) => name.includes('mongodb.mongodb-vscode'));
+
+      this.messageBus.emit('mongosh-editor:read-vscode-extensions-done', {
+        vscodeDir: this._vscodeDir,
+        hasMongodbExtension
+      });
+
+      return hasMongodbExtension ? 'mongodb' : 'js';
+    } catch (error) {
+      this.messageBus.emit('mongosh-editor:read-vscode-extensions-failed', {
+        action: 'mongosh-editor:read-vscode-extensions-failed',
+        error: (error as Error).message
+      });
+
+      return 'js';
+    }
+  }
+
+  async _getEditor(): Promise<string|null> {
+    // Check for an external editor in the mongosh configuration.
+    let editor: string | null = await this._instanceState.shellApi.config.get('editor');
+
+    // Check for an external editor in the environment variable.
+    if (!editor && process.env.EDITOR) {
+      editor = process.env.EDITOR;
+    }
+
+    return editor;
+  }
+
+  async _createTempFile({ content, ext }: { content: string; ext: string }): Promise<string> {
+    const tmpDoc = path.join(this._tmpDir, `edit-${new bson.ObjectId()}.${ext}`);
+
+    // Create a temp file to store content that is being edited.
+    await fs.mkdir(path.dirname(tmpDoc), { recursive: true, mode: 0o700 });
+    await fs.writeFile(tmpDoc, beautify(content));
+
+    return tmpDoc;
+  }
+
+  async _readTempFile(tmpDoc: string): Promise<string> {
+    const content = await fs.readFile(tmpDoc, 'utf8');
+    await fs.unlink(tmpDoc);
```

Comment:
```suggestion
    await fs.unlink(tmpDoc);
    this._content = content;
```

(right? this was missing before, wasn’t it?)

Reply:
Missed this after refactoring, was here before and will bring it back)